src/robotlibcore.py

```diff
@@ -174,6 +175,34 @@ def __join_defaults_with_types(self, method, types):
                 types[name] = type(value)
         return types
 
+    def get_keyword_source(self, keyword_name):
+        method = self.__get_keyword(keyword_name)
+        path = self.__get_keyword_path(method)
+        line_number = self.__get_keyword_line(method)
+        if path and line_number:
+            return '%s:%s' % (path, line_number)
+        if path:
+            return path
+        if line_number:
+            return ':%s' % line_number
+        return None
+
+    def __get_keyword_line(self, method):
+        try:
+            source, line_number = inspect.getsourcelines(method)
+        except (OSError, IOError, TypeError):
+            return None
+        for line in source:
+            if line.strip().startswith('def'):
+                return line_number
+            line_number += 1
```

Comment:
Hmm, this doesn't work correctly if someone has a library like this:

```python
class LambdaLib:
    kw = lambda self: 42
```

Above is obviously rather strange but from RF point of view it's valid. I see two solutions:

- Return the original `line_number` if `def` line isn't found at all.
- Just return whatever `getsourcelines` returns.

Reply:
A bigger problem is that decorators returning different function than the original aren't supported correctly. Same issue on Robot side that the moment:
https://github.com/robotframework/robotframework/issues/3516#issuecomment-605635675